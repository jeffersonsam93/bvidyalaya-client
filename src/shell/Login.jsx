import React, { Component } from "react";
import PropTypes from "prop-types";
import { push } from "react-router-redux";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import cx from "classnames";
import logo from "../assets/logo.png";

import { authSuccess } from "./state/actions";

import { withStyles } from "@material-ui/core/styles";
import { Card, CardActions, CardContent, CardMedia } from "@material-ui/core";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import { InputAdornment } from "@material-ui/core";
import AccountCircle from "@material-ui/icons/AccountCircle";
import LockIcon from "@material-ui/icons/Lock";
import userSample from "__mocks__/users";

const styles = theme => ({
  card: {
    maxWidth: 350,
    height: 350,
    position: "absolute",
    margin: "-175px 0 0 -175px",
    left: "50%",
    top: "50%"
  },
  cardError: {
    height: 370,
    margin: "-185px 0 0 -175px"
  },
  cardPopup: {
    maxWidth: 275
    //height: 350
    //position: "absolute",
    //margin: "-175px 0 0 -175px",
    //left: "50%",
    //top: "50%"
  },
  cardErrorPopup: {
    //height: 370
    //margin: "-185px 0 0 -175px"
  },
  media: {
    height: 60,
    margin: "12px 5px",
    backgroundSize:'100px',
  },
  margin: {
    margin: theme.spacing.unit
  },
  actions: {
    display: "flex"
  },
  right: {
    marginLeft: "auto",
    marginRight: "10px"
  }
});

class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      logInFailed: false,
      isLoaded: false,
      username: "",
      password: ""
    };
    if (process.env.NODE_ENV !== "production") {
      this.state.username = userSample[0].name;
      this.state.password = userSample[0].password;
    }
    this.handleLoginClick = this.handleLoginClick.bind(this);
    this.handleInputChange = this.handleInputChange.bind(this);
  }

  handleLoginClick(event) {
    const { username, password } = this.state;
    const user = userSample.find(element => {
      return element.name.toLowerCase() === username.toLowerCase();
    });

    if (user && user.password === password) {
      this.setState({ logInFailed: false });
      this.props.login({ name: user.name, fullName: user.fullName });
    } else {
      this.setState({ logInFailed: true });
    }
    event.preventDefault();
    return false;
  }

  //componentDidMount() {
  // fetch("assets/config.json")
  //   .then(res => res.json())
  //   .then(
  //     result => {
  //       this.setState({
  //         isLoaded: true,
  //         apiService: result.apiService
  //       });
  //     },
  //     // Note: it's important to handle errors here
  //     // instead of a catch() block so that we don't swallow
  //     // exceptions from actual bugs in components.
  //     error => {
  //       this.setState({
  //         isLoaded: true,
  //         error
  //       });
  //       console.error(error);
  //     }
  //   );
  //}

  handleInputChange(event) {
    const target = event.target;
    const value = target.type === "checkbox" ? target.checked : target.value;
    const name = target.name;
    if (this.state[name] !== value) {
      this.setState({ [name]: value });
      if (this.state.logInFailed) {
        this.setState({ logInFailed: false });
      }
    }
  }

  render() {
    const { classes, popup } = this.props;
    const { logInFailed /*, isLoaded*/ } = this.state;
    //const { from } = this.props.location.state || { from: { pathname: "/" } };
    // if (!isLoaded) {
    //   return null;
    // }
    const cardClass = cx({
      [" " + classes.card]: !popup,
      [" " + classes.cardPopup]: popup,
      [" " + classes.cardError]: logInFailed && !popup,
      [" " + classes.cardErrorPopup]: logInFailed && popup
    });

    return (
      <form onSubmit={this.handleLoginClick}>
        <Card className={cx(cardClass)}>
          {!popup && (
            <div style={{ display:'flex' }}>
              <img className={classes.media} src={logo} title="BVPS" />
              <Typography
              variant="h6"
              style={{ textAlign: "center",margin:"25px 0px" }}
              >
                Bharathi Vidyalaya
              </Typography>
            </div>
          )}
          {logInFailed && (
            <Typography
              variant="subheading"
              color="error"
              style={{ textAlign: "center" }}
            >
              Login failed
            </Typography>
          )}
          <CardContent>
            <TextField
              className={classes.margin}
              name="username"
              label="Username"
              onChange={this.handleInputChange}
              value={this.state.username}
              InputProps={{
                startAdornment: (
                  <InputAdornment position="start">
                    <AccountCircle />
                  </InputAdornment>
                )
              }}
            />

            <TextField
              className={classes.margin}
              name="password"
              label="Password"
              type="password"
              onChange={this.handleInputChange}
              value={this.state.password}
              InputProps={{
                startAdornment: (
                  <InputAdornment position="start">
                    <LockIcon />
                  </InputAdornment>
                )
              }}
            />
          </CardContent>
          <CardActions className={classes.actions} disableActionSpacing>
            <Button
              className={classes.right}
              variant="raised"
              type="submit"
              size="small"
              onSubmit={this.handleLoginClick}
              color="primary"
            >
              Login
            </Button>
            {/* <Button size="small" color="primary">
              Learn More
          </Button> */}
          </CardActions>
        </Card>
      </form>
    );
  }
}

Login.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withRouter(
  connect(null, dispatch => ({
    login: usr => {
      dispatch(authSuccess(usr));
      dispatch(push("/configuration"));
    }
  }))(withStyles(styles)(Login))
);
