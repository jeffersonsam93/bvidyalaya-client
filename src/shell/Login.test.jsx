import React from "react";
import renderer from "react-test-renderer";
import TestWrapper from "testing";
import Login from "./Login";

describe("Login", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <TestWrapper>
          <Login />
        </TestWrapper>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
