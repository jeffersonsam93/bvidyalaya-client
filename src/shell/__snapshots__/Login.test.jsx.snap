// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="MuiPaper-root-10 MuiPaper-elevation2-14 MuiPaper-rounded-11 MuiCard-root-9  Login-card-1"
  >
    <div
      className="MuiCardMedia-root-37 Login-media-5"
      src={undefined}
      style={
        Object {
          "backgroundImage": "url(logoeagle.svg)",
        }
      }
      title="Eagle"
    />
    <div
      className="MuiCardContent-root-39"
    >
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-40 Login-margin-6"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-49 MuiInputLabel-root-44 MuiInputLabel-formControl-45 MuiInputLabel-animated-48 MuiInputLabel-shrink-47"
          data-shrink={true}
          htmlFor={undefined}
        >
          Username
        </label>
        <div
          className="MuiInput-root-54 MuiInput-formControl-55 MuiInput-underline-58"
        >
          <div
            className="MuiInputAdornment-root-67 MuiInputAdornment-positionStart-68"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-70"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g>
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </g>
            </svg>
          </div>
          <input
            aria-invalid={false}
            aria-required={false}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-62"
            defaultValue={undefined}
            disabled={false}
            id={undefined}
            name="username"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="text"
            value="Eagle"
          />
        </div>
      </div>
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-40 Login-margin-6"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-49 MuiInputLabel-root-44 MuiInputLabel-formControl-45 MuiInputLabel-animated-48 MuiInputLabel-shrink-47"
          data-shrink={true}
          htmlFor={undefined}
        >
          Password
        </label>
        <div
          className="MuiInput-root-54 MuiInput-formControl-55 MuiInput-underline-58"
        >
          <div
            className="MuiInputAdornment-root-67 MuiInputAdornment-positionStart-68"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-70"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g>
                <path
                  d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                />
              </g>
            </svg>
          </div>
          <input
            aria-invalid={false}
            aria-required={false}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-62 MuiInput-inputType-65"
            defaultValue={undefined}
            disabled={false}
            id={undefined}
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="password"
            value="eagle"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiCardActions-root-76 Login-actions-7"
    >
      <button
        className="MuiButtonBase-root-93 MuiButton-root-78 MuiButton-raised-83 MuiButton-raisedPrimary-84 MuiButton-sizeSmall-90 Login-right-8"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onSubmit={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="submit"
      >
        <span
          className="MuiButton-label-79"
        >
          Login
        </span>
        <span
          className="MuiTouchRipple-root-95"
        />
      </button>
    </div>
  </div>
</form>
`;
